discord_bot/discord_bot.py

```diff
@@ -34,53 +36,94 @@ def load_config():
+        Sends a heartbeat payload every heartbeat interval and allows messages to be sent.
         """
         await asyncio.sleep(self.heartbeat_interval_ms / 1000.0)
+        print("Last Sequence: {}".format(self.last_seq))
         asyncio.ensure_future(self.send_json({"op": Opcodes.HEARTBEAT, "d": self.last_seq}))
         asyncio.ensure_future(self.heartbeat())
+        # asyncio.ensure_future(self.send_message("embed": {
+        #                                             "title": ("{data[repository][owner_name]}/"
+        #                                                       "{data[repository][name]} "
+        #                                                       "{data[status_message]}"
+        #                                                       ).format(data=data),
+        #                                             "type": "rich",
+        #                                             "description": ("{data[author_name]} {data[type]} "
+        #                                                             "<{data[compare_url]}>"
+        #                                                             ).format(data=data),
+        #                                             "url": data['build_url']}))
 
     async def gateway_handler(self):
         """
-        Creates the websocket, receives messages and acts on them.
+        Creates the websocket, receives responses and acts on them.
         """
         async with websockets.connect("{}/?v={}&encoding={}".format(self.gateway_ws_url,
                                                                     self.config["gateway_api_version"],
                                                                     self.config["gateway_encoding"])) as websocket:
             self.websocket = websocket
             while True:
-                message = await self.websocket.recv()
-                message = json.loads(message)
-                print("{}: {}".format(Opcodes(message["op"]).name, message))
-                if message["op"] == Opcodes.HELLO:
-                    asyncio.ensure_future(self.handshake(message))
-                elif message["op"] == Opcodes.DISPATCH:
+                response = await self.websocket.recv()
+                response = json.loads(response)
+                print("{}: {}".format(Opcodes(response["op"]).name, response))
+                if response["op"] == Opcodes.HELLO:
+                    asyncio.ensure_future(self.handshake(response))
+                elif response["op"] == Opcodes.HEARTBEAT_ACK:
                     pass
-                elif not message["s"]:
-                    self.last_seq = None
+                elif response["op"] == Opcodes.INVALID_SESSION:
+                    print('Invalid Session')
+                elif response["op"] == Opcodes.DISPATCH:
+                    self.last_seq = response["s"]
+                    event = response["t"]
+                    if event == 'READY':
```

Comment:
No magic constants! We should define the `READY` string as a constant for different `DISPATCH` event types.